### PR TITLE
Updated IE_EndUserAddress to support IPv6 field

### PR DESCRIPTION
**Pull request by decastromonteiro**
Added the ConditionalField PDPAddress_IPv6 and changed the IPField name from PDPAddress to PDPAddress_IPv4.
Their condition matches 3GPP TS 29.060 7.7.27 Topic, which states that if there is only IPv4, length must be 6, if there is only IPv6, length must be 18 and if there is IPv4v6, length must be 22.

Also added new tests for this modification under IE_EndUserAddress() basic instatiations.

### CODE REVIEW

File: scapy/contrib/gtp.py [hunk 1]
```diff
@@ -193,13 +193,13 @@ def answers(self, other):
...
                 self.payload.answers(other.payload))
 
-    @classmethod
-    def dispatch_hook(cls, _pkt=None, *args, **kargs):
-        if _pkt and len(_pkt) >= 1:
-            if (struct.unpack("B", _pkt[0])[0] >> 5) & 0x7 == 2:
-                import gtp_v2
-                return gtp_v2.GTPHeader
-        return GTPHeader
+    @classmethod
```
Comment (gpotter2) [inline scapy/contrib/gtp.py]: Looks like a mistake

**Comment by decastromonteiro** [comment]
Well, I don't know why ^M keeps showing up on the file, when I go editing it, it doesn't show up, can someone help?

File: scapy/contrib/gtp.py (hunk 1 above)
Comment (decastromonteiro) [inline scapy/contrib/gtp.py]: Well, I don't know why ^M keeps showing up on the file, when I go editing it, it doesn't show up, any suggestions?

**Comment by gpotter2** [comment]
- Start vim
- do `:set ff=unix`
- do `:wq`

Should fix it

File: scapy/contrib/gtp.uts [hunk 2]
```diff
@@ -146,12 +146,17 @@ ie.ietype == 127 and ie.Charging_id == 0xff1bc3f2
...
 
 = IE_EndUserAddress(), basic instantiation
 ie = IE_EndUserAddress(
-    length=6, PDPTypeOrganization=1, PDPTypeNumber=0x21, PDPAddress='10.42.0.3')
-ie.ietype == 128 and ie.length == 6 and ie.PDPTypeOrganization == 1 and ie.PDPTypeNumber == 0x21 and ie.PDPAddress == '10.42.0.3'
+    length=22, PDPTypeOrganization=1, PDPTypeNumber=0x8d, PDPAddress_IPv4='10.42.0.3', PDPAddress_IPv6 = '2001:db8:a0b:12f0::1')
+ie.ietype == 128 and ie.length == 22 and ie.PDPTypeOrganization == 1 and ie.PDPTypeNumber == 0x8d and ie.PDPAddress_IPv4 == '10.42.0.3' and ie.PDPAddress_IPv6 == '2001:db8:a0b:12f0::1'
+
+= IE_EndUserAddress(), basic instantiation
+ie = IE_EndUserAddress(
```
Comment (gpotter2) [inline scapy/contrib/gtp.uts]: Could you set a different name for this test ?

File: scapy/contrib/gtp.py [hunk 3]
```diff
@@ -192,14 +192,15 @@ def answers(self, other):
...
-    def dispatch_hook(cls, _pkt=None, *args, **kargs):
-        if _pkt and len(_pkt) >= 1:
-            if (struct.unpack("B", _pkt[0])[0] >> 5) & 0x7 == 2:
-                import gtp_v2
-                return gtp_v2.GTPHeader
-        return GTPHeader
+    
+    
+    @classmethod
+    def dispatch_hook(cls, _pkt=None, *args, **kargs):
```
Comment (gpotter2) [inline scapy/contrib/gtp.py]: Also, please try to fix the line ending issue

File: scapy/contrib/gtp.py [hunk 4]
```diff
@@ -342,8 +343,10 @@ class IE_EndUserAddress(IE_Base):
                     BitField("SPARE", 15, 4),
                     BitField("PDPTypeOrganization", 1, 4),
                     XByteField("PDPTypeNumber", None),
-                    ConditionalField(IPField("PDPAddress", RandIP()),
-                                     lambda pkt: pkt.length > 2)]
+                    ConditionalField(IPField("PDPAddress_IPv4", RandIP()),
+                                     lambda pkt: pkt.length == 6 or pkt.length == 22),
+                    ConditionalField(IP6Field("PDPAddress_IPv6", "::1"), lambda
+                                    pkt: pkt.length == 18 or pkt.length == 22)]
```
Comment (gpotter2) [inline scapy/contrib/gtp.py]: Could you move the lambda keyword to the line below (more readable) ?

File: scapy/contrib/gtp.py [hunk 5]
```diff
@@ -192,14 +192,15 @@ def answers(self, other):
...
                 self.payload.answers(other.payload))
-
-    @classmethod
-    def dispatch_hook(cls, _pkt=None, *args, **kargs):
-        if _pkt and len(_pkt) >= 1:
-            if (struct.unpack("B", _pkt[0])[0] >> 5) & 0x7 == 2:
-                import gtp_v2
-                return gtp_v2.GTPHeader
-        return GTPHeader
+
```
Comment (p-l-) [inline scapy/contrib/gtp.py]: If you are not changing the code here, please do not mess up with whitespaces as it breaks code history for no reason.

File: scapy/contrib/gtp.py (hunk 4 above)
Comment (p-l-) [inline scapy/contrib/gtp.py]: Agreed. Also, why choose `"::1"` as default value when the IPv4 counterpart has `RandIP()`?

File: scapy/contrib/gtp.uts (hunk 2 above)
Comment (p-l-) [inline scapy/contrib/gtp.uts]: Just add 'IPv6'

File: scapy/contrib/gtp.py (hunk 5 above)
Comment (decastromonteiro) [inline scapy/contrib/gtp.py]: Hey, I didn't change the code on the @classmethod, I only altered the IE_EndUserAddress() part, but when I saved the file that ^M which appeared. I edited it several times now trying to remove it, but was not able. Not in pycharm, vim, nano or even Sublime Text. I'm out of ideas on how to remove that ^M from the @classmethod.

All in all, I added more descriptions to the gtp.uts added tests, I've put the lambda expressions on the same line, and finally added RandIP6() to the IPv6 field.

Thanks for you feedback.

Comment (gpotter2) [inline scapy/contrib/gtp.py]: We can't leave a ^M in the code, it's really ugly and may crash with some python versions...
Have you tried https://github.com/secdev/scapy/pull/531#issuecomment-282588778 ?

Comment (decastromonteiro) [inline scapy/contrib/gtp.py]: Thanks @gpotter2 , just followed the  http://aruljohn.com/info/ctrlm/ and it worked, but as every line did have a ^M character, it seems like I have altered the whole document... hope that's OK.

All tests passed.

Thanks for your feedback and support.

**Comment by decastromonteiro** [comment]
Dear,

Had to rollback the RandIP6() modification because it seems to break the packet.show().

See the traceback I got:

gtp_packet.IE_list.append(IE_EndUserAddress(length=22, PDPTypeNumber=0x8d))
gtp_packet.show()

_Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 898, in show
    return self._show_or_dump(dump, indent, lvl, label_lvl)
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 889, in _show_or_dump
    s += self.payload._show_or_dump(dump=dump, indent=indent, lvl=lvl+(" "*indent*self.show_indent), label_lvl=label_lvl, first_call=False)
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 889, in _show_or_dump
    s += self.payload._show_or_dump(dump=dump, indent=indent, lvl=lvl+(" "*indent*self.show_indent), label_lvl=label_lvl, first_call=False)
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 889, in _show_or_dump
    s += self.payload._show_or_dump(dump=dump, indent=indent, lvl=lvl+(" "*indent*self.show_indent), label_lvl=label_lvl, first_call=False)
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 889, in _show_or_dump
    s += self.payload._show_or_dump(dump=dump, indent=indent, lvl=lvl+(" "*indent*self.show_indent), label_lvl=label_lvl, first_call=False)
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 876, in _show_or_dump
    s += fvalue._show_or_dump(dump=dump, indent=indent, label_lvl=label_lvl+lvl+"   |", first_call=False)
  File "/usr/local/lib/python2.7/dist-packages/scapy/packet.py", line 881, in _show_or_dump
    reprval = f.i2repr(self,fvalue)
  File "/usr/local/lib/python2.7/dist-packages/scapy/layers/inet6.py", line 228, in i2repr
    if in6_isaddrTeredo(x):   # print Teredo info
  File "/usr/local/lib/python2.7/dist-packages/scapy/utils6.py", line 474, in in6_isaddrTeredo
    our = inet_pton(socket.AF_INET6, x)[0:4]
  File "/usr/local/lib/python2.7/dist-packages/scapy/pton_ntop.py", line 79, in inet_pton
    return socket.inet_pton(af, addr)
TypeError: inet_pton() argument 2 must be string, not instance_

**Pull request closed without merging**